secrets/secrets.go

```diff
@@ -0,0 +1,270 @@
+package secrets
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/SUSE/scf-secret-generator/model"
+	"github.com/SUSE/scf-secret-generator/password"
+	"github.com/SUSE/scf-secret-generator/ssh"
+	"github.com/SUSE/scf-secret-generator/ssl"
+	"github.com/SUSE/scf-secret-generator/util"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// The unversioned name of the secret used by legacy versions of the secrets generator
+const legacySecretName = "secret"
```

Comment:
_Maybe_ call this `kLegacySecretName` so it's (slightly) more obvious it's a ċonstant… not sure how I feel about that